### DISCUSSION TITLE
cmd replacement

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#3228] [Help] [answered]
Hello,

I'm currently using the windows batch language  and have scripts with a quite complex controll flow to determine witch templates are combined to the final image. These scripts are getting unreadable . Thats the reason why I would like to replace the windows shell by a csh like shell, which has better structures like switch statements etc. Howoever I didn't find a csh implementation for windows and I would like to avoid the linux subsystem because I have to run even bat files.

Does anyone know a native windows shell with a language similar to C or Pascal?

**Comment by snibgo** [comment, marked as answer]
tcsh is available from Cygwin.
